src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -641,35 +641,13 @@ internal static bool IsCapturedPrimaryConstructorParameter(BoundExpression expre
                    primaryCtor.GetCapturedParameters().ContainsKey(parameter);
         }
 
-        private enum ReceiverCaptureMode
-        {
-            /// <summary>
-            /// No special capture of the receiver, unless arguments need to refer to it.
-            /// For example, in case of a string interpolation handler.
-            /// </summary>
-            Default = 0,
-
-            /// <summary>
-            /// Used for a regular indexer compound assignment rewrite.
-            /// Everything is going to be in a single setter call with a getter call inside its value argument.
-            /// Only receiver and the indexes can be evaluated prior to evaluating the setter call. 
-            /// </summary>
-            CompoundAssignment,
-
-            /// <summary>
-            /// Used for situations when additional arbitrary side-effects are possibly involved.
-            /// Think about deconstruction, etc.
-            /// </summary>
-            UseTwiceComplex
-        }
-
         /// <summary>
         /// Visits all arguments of a method, doing any necessary rewriting for interpolated string handler conversions that
         /// might be present in the arguments and creating temps for any discard parameters.
         /// </summary>
         private ImmutableArray<BoundExpression> VisitArgumentsAndCaptureReceiverIfNeeded(
             [NotNullIfNotNull(nameof(rewrittenReceiver))] ref BoundExpression? rewrittenReceiver,
-            ReceiverCaptureMode captureReceiverMode,
+            bool forceReceiverCapturing,
```

Comment:
Consider documenting this parameter, similarly to how the enum was previously documented.